File: c/exercise13/compound_interest.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
  float principal = 0.0;
  printf("What is the principal amount? ");
  scanf("%f", &principal);

  float rate = 0.0;
  printf("What is the rate? ");
  scanf("%f", &rate);

  int years = 0;
  printf("What is the number of years? ");
  scanf("%d", &years);

  int compound_years = 0;
  printf("What is the number of times the interest is compounded per year? ");
  scanf("%d", &compound_years);

  float investment = principal * ((1 + (rate / (float)compound_years)) *
                                  (float)compound_years * years);

  printf("$%.2f invested at %.2f%% for %d years\ncompounded %d times per year "
         "is $%.2f.\n",
         principal, rate, years, compound_years, investment);

  return EXIT_SUCCESS;
}

File: python/exercise13/compound_interest.py
principal = int(input("What is the principal amount? "))
rate = float(input("What is the rate? "))
years = int(input("What is the number of years? "))
compound_years = int(
    input("What is the number of times the interest is compounded per year? ")
)

investment = float(principal) * (
    (1 + ((rate / float(100)) / float(compound_years))) * float(compound_years * years)
)

print(
    f"${principal} invested at {rate:.2f} for {years} years\ncompounded {compound_years} times per year is ${investment:.2f}."
)
